yt_dlp/extractor/freetv.py

```diff
@@ -0,0 +1,186 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    ExtractorError,
+    try_get,
+    urlencode_postdata, int_or_none, str_or_none,
+)
+
+
+class FreeTvBaseIE(InfoExtractor):
+    def _get_api_response(self, content_id, postdata, resource_type):
+        response = self._download_json(
+            'https://www.freetv.com/wordpress/wp-admin/admin-ajax.php',
+            content_id, data=urlencode_postdata(postdata), fatal=False,
+            note='Downloading {} {} JSON'.format(content_id, resource_type))
+
+        if response is False:
+            raise ExtractorError("Couldn't get response", expected=True)
+
+        if response.get('data') is False:
+            raise ExtractorError("Response doesn't contain {} data".format(resource_type), expected=True)
+
+        return response
+
+    def _extract_video(self, content_id, action="olyott_video_play"):
+        api_response = self._get_api_response(
+            content_id, resource_type='video',
+            postdata={
+                'action': action,
+                'contentID': content_id,
+            })
+
+        video_id = try_get(api_response, lambda x: x['data']['displayMeta']['contentID'], expected_type=str)
+        title = try_get(api_response, lambda x: x['data']['displayMeta']['title'])
+        description = try_get(api_response, lambda x: x['data']['displayMeta']['desc'])
+        video_url = try_get(api_response, lambda x: x['data']['displayMeta']['streamURLVideo'], expected_type=str)
```

Comment:
FYI, `try_get(A, lambda x: x[b][c][d])` can be written as `traverse_obj(A, (b, c, d))`. Both are fine, just letting u know since `traverse_obj` is generally easier